src/gui/wizard/owncloudwizard.cpp

```diff
@@ -65,6 +67,22 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
     setPage(WizardCommon::Page_Result, _resultPage);
     setPage(WizardCommon::Page_WebView, _webViewPage);
 
+    // Add keyboard shortcut to allow the user to quit
+    auto *closeWindowAction = new QAction(this);
+    closeWindowAction->setShortcut(QKeySequence("Ctrl+W"));
+    connect(closeWindowAction, &QAction::triggered, this, &QApplication::quit);
+    addAction(closeWindowAction);
+
+    // Enable "Cancel" button; reject() is set to quit()
+    QWizard::setOptions(QWizard::CancelButtonOnLeft);
+
+    // Make the application quit when the window is closed
+    // This setting will be reversed when the wizard is complete
+    QApplication::setQuitOnLastWindowClosed(true);
```

Comment:
@elsiehupp I'd say we also need to check a condition before setting it to **true**. If there is an account already set up, we don't want the Wizard close action leading to the application quitting. Similar to what @er-vin mentioned already for the `reject()` method.

Reply:
It looks like I can add the code `if (AccountManager::instance()->accounts().isEmpty())` for this purpose. I haven't tried it, yet, though.